irispy/obsid.py

```diff
@@ -0,0 +1,152 @@
+
+    Examples
+    --------
+    Quickly show OBS ID parameters:
+
+    >>> obsid.Obsid(3675508564)
+    IRIS OBS ID 3675508564
+    ----------------------
+    Description:            Large dense 96-step raster 31.35x120 96s
+    SJI filters:                                    C II   Mg II w s
+    SJI field of view:                                       120x120
+    Exposure time:                                               8.0 s
+    Binning:                               Spatial x 1, Spectral x 1
+    FUV binning:                         FUV spectrally rebinned x 4
+    SJI cadence:                                 SJI cadence default
+    Compression:                                Lossless compression
+    Linelist:                                       Flare linelist 1
+
+    The data can be accessed as in a dictionary:
+    
+    >>> data = obsid.Obsid(3675508564)
+    >>> data['exptime']
+    <Quantity 8. s>
+    >>> data['linelist']
+    'Flare linelist 1'
+    >>> data['raster_fov']
+    '31.35x120'
+    >>> data['raster_step']
+    0.33
+    """
+
+    def __init__(self, obsid):
+        self.obsid = obsid
+        self.field_keys = {'Large linelist': 'linelist',
+                           'Default compression': 'compression',
+                           'Lossy compression': 'compression',
+                           'Non-simultaneous readout': 'readout',
+                           'SJI cadence default': 'sji_cadence',
+                           'SJI cadence 10s': 'sji_cadence',
+                           'FUV binned same as NUV': 'fuv_binning',
+                           'Spatial x 1, Spectral x 1': 'binning',
+                           'Exposure 1s': 'exptime',
+                           'C II   Si IV   Mg II h/k   Mg II w   ': 'sjis'}
+        self._versions = [36, 38, 40]
+        data, options = self.read_obsid(obsid)
+        super().__init__(data)
+        self.options = options
+
+    def __repr__(self):
+        return ("IRIS OBS ID {obsid}\n"
+                "----------------------\n"
+                "Description:       {raster_fulldesc:>45}\n"
+                "SJI filters:       {sjis:>45}\n"
+                "SJI field of view: {sji_fov:>45}\n"
+                "Exposure time:     {exptime:>45}\n"
+                "Binning:           {binning:>45}\n"
+                "FUV binning:       {fuv_binning:>45}\n"
+                "SJI cadence:       {sji_cadence:>45}\n"
+                "Compression:       {compression:>45}\n"
+                "Linelist:          {linelist:>45}").format(**self)
+
+    @staticmethod
+    def __exptime_to_quant(exptime):
+        """
+        Converts an 'exptime' string (used in IRIS tables and OBS_DESC)
+        to a Quantity instance in seconds.
+        """
+        if exptime == "Exposure 1s":
+            return 1. * u.s
+        else:
+            return float(exptime.split(' x ')[1]) * u.s
+
+    def read_obsid(self, obsid):
```

Comment:
This method should be hidden (```_read_obsid```) since, as I understand, this function is not supposed to be used directly by users, but rather called by the ```__init__```.  Also making it hidden means we give the user no guarantee that its API wont change in the future.  So if we want to rewrite it for whatever reason, we don't have to worry about deprecation warnings etc.  Easier for maintenance.